pkg/server/proto/server.proto

```diff
@@ -1087,6 +1105,35 @@ message TaskLaunchInfo {
   repeated string arguments = 3;
 }
 
+// Task tracks the life of an on-demand runner task that spawns Start
+// and Stop tasks for any kind of job/operation in Waypoint. Automatic jobs
+// such as project polling or status report generation spawn on-demand
+// runner tasks, and this message can be used to track the life of those
+// automated jobs. Note that every on-demand runner task is wrapped with a
+// Start and Stop task, which we track here as well.
+message Task {
```

Comment:
I also just thought, we're probably going to need some form of task expiry. You can get a lot of the timestamps from the jobs, but there is probably a state API we'll eventually need. You don't need to do anything in this PR I just wanted to put it in your head so you're aware of it and think about it as the implementation progresses.

Reply:
I think my concern with this is likely in the upper layer, mostly figuring out when TaskDelete is going to be called. The Task api used doesn't mandate an explicit place for Tasks to be finished, so I'd like to understand how we'll prevent Task values from building up over time.

Reply:
I think they'll be part of the pruning similar to jobs (not sure how you did that in the sql-side). Otherwise, I think we set a timeout currently on ODR tasks so we can just double or triple that (for a really safe buffer) prior to deleting the task. But yes, this is a really important thing to look closely at @briancain when you do the implementation.

Reply:
I wouldn't just add pruning, I'd think about how the lifecycle relates to it since tasks are tied to jobs, unlike jobs which stand on their own. Pruning _may_ be the answer, but I think we can potentially have a more solid solution.

Reply:
Makes sense...I should of hit refresh before sending that comment! I missed the first one after Evans.

I can see us cleaning up tasks on ODR timeouts for sure, that use-case makes sense to me 👍🏻 Perhaps I can write up a little gist idea to chat about in slack for a plan going forward. I can foresee this being addressed in Phase 3 of the plan, where we implement actually storing these Task messages on ODR job queues in the server and later in the runner on execution.

Reply:
I added a TODO above the Task message docs for me for phase 3! I think that should cover it until I get there later after the API/CLI phase 2.